security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -48,6 +49,11 @@ var (
 	newFileWatcher = filewatcher.NewWatcher
 	// The total timeout for any credential retrieval process, default value of 10s is used.
 	totalTimeout = time.Second * 10
+
+	// UseRotationWithCert is a field controlling the use the cert for rotation
+	// If is set to true, we will use the old cert to rotate instead of token
+	UseRotationWithCert = env.RegisterBoolVar("USE_ROTATION_WITH_CERT", false,
```

Comment:
Flags should go to pilot-agent/main, be passed via Option

Reply:
Ok. So you mean you first waiting for this change https://github.com/istio/istio/pull/2536 to be committed?  I think the PROV_CERT also use the environment variable

Reply:
This is not the right place to declare env.

The env name is also not clear

Reply:
There is an issue about refactoring all the env name. As it is a big and thorough change, we will do that in the next release